x-pack/plugins/ingest_pipelines/public/application/services/navigation.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const BASE_PATH = '/';
+
+const EDIT_PATH = 'edit';
+
+const CREATE_PATH = 'create';
+
+const _getEditPath = (name: string, encode = true): string => {
+  return `${BASE_PATH}${EDIT_PATH}/${encode ? encodeURIComponent(name) : name}`;
+};
+
+const _getCreatePath = (): string => {
+  return `${BASE_PATH}${CREATE_PATH}`;
+};
+
+const _getClonePath = (name: string, encode = true): string => {
+  return `${BASE_PATH}${CREATE_PATH}/${encode ? encodeURIComponent(name) : name}`;
+};
+const _getListPath = (name?: string): string => {
+  return `${BASE_PATH}${name ? `?pipeline=${encodeURIComponent(name)}` : ''}`;
+};
+
+export const ROUTES_CONFIG = {
+  list: _getListPath(),
+  edit: _getEditPath(':name', false),
+  create: _getCreatePath(),
+  clone: _getClonePath(':sourceName', false),
+};
+
+export const getListPath = ({
```

Comment:
Nit: what do we gain by exporting adapter functions instead of the original functions? For example:

```js
// Setting a default for `encode` supports the simpler consumption pattern of not 
// having to specify it, and I don't think we lose anything by allowing the consumer
// to specify `false`.
export const getEditPath = ({ pipelineName, encode = true }: { pipelineName: string, encode?: boolean }): string => {
  return `${BASE_PATH}${EDIT_PATH}/${encode ? encodeURIComponent(name) : name}`;
};

/* snip */

export const ROUTES_CONFIG = {
  edit: getEditPath({ pipelineName: ':name', encode: false }),
  /* snip */
};
```

Reply:
I think that encoding the parameter is an implementation detail and should not be needed for clients using the navigation service, hence a separate `ROUTES` constant.